Python/feature_extraction.py

```diff
@@ -230,43 +232,41 @@ def prepare_batch_data(feature_list, labels_list, is_training=True):
                 for k in mid_segment_frames_idx:
                     for l in range(k - label_smearing, k + label_smearing + 1):
 
-                        if padding_length <= l < num_beats - padding_length:
+                        if padding_length <= l < num_beats + padding_length:
 
                             next_window = features[:, l-padding_length: l+padding_length+1]
-                            next_label = 0
-                            next_weight = 1
 
                             data_x[feature_count, :, :] = next_window
-                            data_y[feature_count] = next_label
-                            data_weight[feature_count] = next_weight
+                            data_y[feature_count] = 0
+                            data_weight[feature_count] = 1
                             track_idx[feature_count] = current_track
 
                             feature_count += 1
 
             # sample randomly from the remaining frames
-            remaining_frames_idx = [i for i in range(num_beats) if (i not in positive_frames_idx)]
+            remaining_frames_idx = []
+            for i in range(features.shape[1]):
+                if (i not in positive_frames_idx) and (padding_length <= i < features.shape[1] - padding_length):
+                    remaining_frames_idx.append(i)
```

Comment:
here we were mixing up `num_beats` (unpadded feature count) with the various `frames_idx` (padded features) -- I think this just meant we weren't getting any negative frames from the later part of files